### PR TITLE
Add support for prometheus scraper

### PR DESCRIPTION
**Pull request by oded-dd**

### CODE REVIEW

File: eks/aws-node-termination-handler/Settings.dhall [hunk 1]
```diff
@@ -62,45 +72,74 @@ let Knobs =
...
+                { enable : Bool
+                , port : Text
+                , interval : Text
+                , endpoint : Text
+                }
+            , default =
+                { enable = True
+                , port = "9092"
+                , interval = "30s"
+                , endpoint = "kube-prometheus"
```
[the remark this answers is not in the document]
Comment (oded-dd) [reply eks/aws-node-termination-handler/Settings.dhall:113]: Complicates stuff, ignoring

File: eks/aws-node-termination-handler/Settings.dhall [hunk 2]
```diff
@@ -62,45 +72,74 @@ let Knobs =
...
 
       in  { Type = { webhook : Webhook.Type }
           , default.webhook = Webhook.default
           , Webhook = Webhook
           }
 
+let Monitoring =
+      let Prometheus =
+            { Type =
+                { enable : Bool
```
[the remark this answers is not in the document]
Comment (oded-dd) [reply eks/aws-node-termination-handler/Settings.dhall]: I've set it to be `operator : Optional { interval : Text }`

File: eks/aws-node-termination-handler/kubernetes/PrometheusRules.dhall [hunk 3]
```diff
@@ -0,0 +1,22 @@
+let imports = ../../../imports.dhall
+
+let Kubernetes = imports.Kubernetes
+
+let Prometheus = imports.Prometheus
+
+let Common = ./Common.dhall
+
+in    λ(common : Common.Type)
+    → Prometheus.PrometheusRule::{
```
[the remark this answers is not in the document]
Comment (oded-dd) [reply eks/aws-node-termination-handler/kubernetes/PrometheusRules.dhall]: I don't agree. This should be kept simple, the responsibility should be outside this repo, as part of a deployment script

File: eks/aws-node-termination-handler/kubernetes/ServiceMonitor.dhall [hunk 4]
```diff
@@ -0,0 +1,26 @@
+let imports = ../../../imports.dhall
+
+let Kubernetes = imports.Kubernetes
+
+let Prometheus = imports.Prometheus
+
+let Common = ./Common.dhall
+
+in    λ(common : Common.Type)
+    → Prometheus.ServiceMonitor::{
```
[the remark this answers is not in the document]
Comment (oded-dd) [reply eks/aws-node-termination-handler/kubernetes/ServiceMonitor.dhall]: I don't agree. This should be kept simple, the responsibility should be outside this repo, as part of a deployment script

File: eks/aws-node-termination-handler/kubernetes/DaemonSet.dhall [hunk 5]
```diff
@@ -192,22 +198,58 @@ in    λ(common : Common.Type)
...
+                                        render.bool
+                                          common.settings.prometheus.enable
+                                    }
+                                  , Kubernetes.EnvVar::{
+                                    , name = "PROMETHEUS_SERVER_PORT"
+                                    , value =
+                                        render.optional-text
+                                          ( Some
+                                              common.settings.prometheus.port
+                                          )
```
Comment (ari-becker) [inline eks/aws-node-termination-handler/kubernetes/DaemonSet.dhall:225]: `value = common.settings.prometheus.port`

File: eks/aws-node-termination-handler/kubernetes/PrometheusRules.dhall [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+let Prometheus = imports.Prometheus
+
+let Common = ./Common.dhall
+
+in  λ(common : Common.Type) →
+      Prometheus.PrometheusRule::{
+      , metadata = common.metadata ⫽ { labels = Some common.labels.prometheus }
+      , spec = Prometheus.PrometheusRuleSpec::{=}
+      }
```
Comment (ari-becker) [inline eks/aws-node-termination-handler/kubernetes/PrometheusRules.dhall:13]: What's the point of including an empty rule set here?

Comment (oded-dd) [reply eks/aws-node-termination-handler/kubernetes/PrometheusRules.dhall:13]: Once we will define the relevant rules it will be added

**Pull request merged**